File: src/Entity/Categorie.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use App\Repository\CategorieRepository;
use Doctrine\DBAL\Types\Types;

#[ORM\Entity(repositoryClass: CategorieRepository::class)]
class Categorie
{
   
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ? int $idCategorie=null;

   
    #[ORM\Column(length : 255)]
    private ? string $nom=null;


    #[ORM\Column(length : 255)]
    private ? string $descriptif=null;

    public function getIdCategorie(): ?int
    {
        return $this->idCategorie;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getDescriptif(): ?string
    {
        return $this->descriptif;
    }

    public function setDescriptif(string $descriptif): self
    {
        $this->descriptif = $descriptif;

        return $this;
    }
    



}

File: AnasHammou/src/tn/gestion/evenement/enitite/Categorie.java
package tn.gestion.evenement.enitite;

public class Categorie {
    private int idCategorie;
    private String nom;
    private String descriptif;

    public Categorie() {
    }

    public Categorie(int idCategorie, String nom, String descriptif) {
        this.idCategorie = idCategorie;
        this.nom = nom;
        this.descriptif = descriptif;
    }

    public Categorie(String nom, String descriptif) {
        this.nom = nom;
        this.descriptif = descriptif;
    }

    public Categorie(int idCategorie) {
        this.idCategorie = idCategorie;
    }

    public int getIdCategorie() {
        return idCategorie;
    }

    public void setIdCategorie(int idCategorie) {
        this.idCategorie = idCategorie;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getDescriptif() {
        return descriptif;
    }

    public void setDescriptif(String descriptif) {
        this.descriptif = descriptif;
    }

    @Override
    public String toString() {
        return "Categorie{" + "idCategorie=" + idCategorie + ", nom=" + nom + ", descriptif=" + descriptif + '}';
    }
    
    
}
